service/src/main/java/org/broadinstitute/listener/config/SamInspectorProperties.java

```diff
@@ -1,3 +1,31 @@
 package org.broadinstitute.listener.config;
 
-public record SamInspectorProperties(String samUrl, String samResourceId) {}
```

Comment:
why not just use `record`?

Reply:
I would have preferred that, but it looks like I can't get the default `samResourceId` set in here if it's a record. Happy to hear better ideas, I'm new to modern Java and may be missing obvious ways to do things.

Reply:
maybe becuz https://github.com/DataBiosphere/terra-azure-relay-listeners/blob/main/service/src/main/resources/application.yml#L18 is not updated?

Btw, I think should update this yml file with the new key...and having the leo specific key as default. We've also been documenting configs in https://github.com/DataBiosphere/terra-azure-relay-listeners#configuration-properties

Reply:
Ah, I got distracted by the nifty environment variables and forgot to update the actual config file. I'll set the default there and revert this to `record`. Will also add information about the sam inspector to the README.